src/core/common/message.cpp

```diff
@@ -148,7 +148,7 @@ ThreadError MessagePool::FreeBuffers(Buffer *aBuffer)
 
 ThreadError MessagePool::ReclaimBuffers(int aNumBuffers)
 {
-    int numFreeBuffers;
+    uint16_t numFreeBuffers;
```

Comment:
With this change aNumBuffers will be promoted to unsigned in the comparison on line 159, I think?  You are welcome to prove me wrong.